lib/remoteproc/remoteproc.c

```diff
@@ -46,6 +46,9 @@ remoteproc_get_mem(struct remoteproc *rproc, const char *name,
 		return NULL;
 
 	metal_list_for_each(&rproc->mems, node) {
+		if(node == 0x0){
+			return NULL;
+		}
```

Comment:
To reproduce the issue seems that rpoc->mems should not initialized. 
I suppose that this issue occurs when rproc structure is not initialized (`remoteproc_init` not called), right?

So here is look like a coding bug, which could be solved by an assert outside the loop.
`metal_assert(!rproc->mems.prev || !rproc->mems.next)`

Reply:
You could also consider modifying the libmetal code, so null pointers / empty lists dont run.
```c
#define metal_list_for_each(list, node)		\
	for ((node) = (list)->next;		\
	     (node) != (list);			\
	     (node) = (node)->next)
	     ```
to 
```c
#define metal_list_for_each(list, node)		\
	for ((node) = (list)->next;		\
	     ((list) != NULL) && ((node) != NULL) && (node) != (list);			\
	     (node) = (node)->next)
	     ```
Im not really familiar enough with libmetal / open-amp to judge the correct approach.
Was taking a stab at making a stand alone elf loader that used the linux devmem interface to get memory from elf file, and got a segfault from not having my remoteproc initialized.
So I most likely was not calling remoteproc_init.